prediction/src/algorithms/evaluation.py

```diff
@@ -0,0 +1,58 @@
+import os
+import sys
+
+sys.path.insert(0, os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', '..'))
+sys.path.append('..')
+import numpy as np
+import pylidc as pl
+from config import Config
+from classify import trained_model
+from tqdm import tqdm
+from src.preprocess.lung_segmentation import get_z_range
+
+CONFIDENCE_THRESHOLD = 0.5
+
+
+def get_accuracy(TP, TN, FP, FN):
+    """
+    The accuracy is the ability to correctly predict the class of an observation.
+    """
+    return (TP + TN) / (TP + TN + FP + FN)
+
+
+def logloss(true_label, predicted, eps=1e-15):
+    """Calculate the logarithmic loss (http://wiki.fast.ai/index.php/Log_Loss)"""
+    p = np.clip(predicted, eps, 1 - eps)
+    if true_label == 1:
+        return -np.log(p)
+    else:
+        return -np.log(1 - p)
+
+
+def evaluate_classification(model_path=None):
+    if model_path is None:
+        model_path = os.path.join(Config.ALGOS_DIR, 'classify', 'assets', 'gtr123_model.ckpt')
+
+    scans = pl.query(pl.Scan).all()
+    confidences = []
+    log_losses = []
+
+    for scan in tqdm(scans[:5]):
+        dicom_path = os.path.join(Config.SMALL_DICOM_PATHS, scan.patient_id, scan.study_instance_uid,
+                                  scan.series_instance_uid)
+        min_z, max_z = get_z_range(dicom_path)
+        nodule_list = []
+        for annotation in scan.annotations:
+            centroid_x, centroid_y, centroid_z = annotation.centroid()
+            z_index = int((centroid_z - min_z) / scan.slice_thickness)
```

Comment:
This step should maybe be transferred to an other location where it can also be tested separately. Computing the slice index from the relative image positions still feels a bit hacky.

Reply:
@Serhiy-Shekhovtsov @vessemer Do you have experience with getting the slice index from the ImagePositient information? The DICOM images have ImagePosition information (z_axis) from e.g. -435 to -63. The nodule has one of -252 so we're required to calculate the slice index (here: 61) out of that. Do you know whether that's the right way or whether we are already doing this somewhere?

Reply:
Is [this](https://github.com/concept-to-clinic/concept-to-clinic/blob/666fb1bf410cbd80eb7928d0e3c5f2433ad668a5/prediction/src/preprocess/load_ct.py#L185) what you are looking for?

Reply:
The offset then used [here](https://github.com/concept-to-clinic/concept-to-clinic/blob/413bc74311579bc3eaa9ba9a147eb8bdc937b093/prediction/src/preprocess/preprocess_ct.py#L167) to convert the coordinates

```
    coord = np.rint((coord - np.array(origin)) * np.array(spacing))
```

Reply:
@WGierke, @Serhiy-Shekhovtsov, I've missed that, it should be `np.rint((coord - np.array(origin)) / np.array(spacing))`, also I didn't get for what reason, @Serhiy-Shekhovtsov, you remove [this](https://github.com/concept-to-clinic/concept-to-clinic/blame/a2b55a863877ac790292969ce072903d552f3abc/prediction/src/preprocess/preprocess_ct.py#L145) line, I should missing something :/